nautobot/dcim/views.py

```diff
@@ -334,6 +348,23 @@ class LocationView(generic.ObjectView):
     queryset = Location.objects.all()
 
     def get_extra_context(self, request, instance):
+        stats = {
+            "rack_count": Rack.objects.restrict(request.user, "view").filter(location=instance).count(),
+            "device_count": Device.objects.restrict(request.user, "view").filter(location=instance).count(),
+            "prefix_count": Prefix.objects.restrict(request.user, "view").filter(location=instance).count(),
+            "vlan_count": VLAN.objects.restrict(request.user, "view").filter(location=instance).count(),
+            "circuit_count": Circuit.objects.restrict(request.user, "view")
+            .filter(terminations__location=instance)
+            .count(),
+            "vm_count": VirtualMachine.objects.restrict(request.user, "view")
+            .filter(cluster__location=instance)
+            .count(),
+        }
+        rack_groups = (
+            RackGroup.objects.add_related_count(RackGroup.objects.all(), Rack, "group", "rack_count", cumulative=True)
+            .restrict(request.user, "view")
+            .filter(location=instance)
+        )
```

Comment:
```suggestion
        all_related_location = [instance.pk, *instance.descendants().values_list("pk", flat=True)]

        stats = {
            "rack_count": Rack.objects.restrict(request.user, "view").filter(location__in=all_related_location).count(),
            "device_count": Device.objects.restrict(request.user, "view")
            .filter(location__in=all_related_location)
            .count(),
            "prefix_count": Prefix.objects.restrict(request.user, "view")
            .filter(location__in=all_related_location)
            .count(),
            "vlan_count": VLAN.objects.restrict(request.user, "view").filter(location__in=all_related_location).count(),
            "circuit_count": Circuit.objects.restrict(request.user, "view")
            .filter(terminations__location__in=all_related_location)
            .count(),
            "vm_count": VirtualMachine.objects.restrict(request.user, "view")
            .filter(cluster__location__in=all_related_location)
            .count(),
        }
        rack_groups = (
            RackGroup.objects.add_related_count(RackGroup.objects.all(), Rack, "group", "rack_count", cumulative=True)
            .restrict(request.user, "view")
            .filter(location__in=all_related_location)
        )
```

Reply:
Reason being, it appears filtering by a Location parent in the filter view includes the children so the numbers don't match up from my experience.

Reply:
```python
>>> print(instance.descendants().query)

    WITH RECURSIVE __tree (
        "tree_depth",
        "tree_path",
        "tree_ordering",
        "tree_pk"
    ) AS (
        SELECT
            0 AS tree_depth,
            array[T.id] AS tree_path,
            array[_name]::text[] AS tree_ordering,
            T."id"
        FROM dcim_location T
        WHERE T."parent_id" IS NULL

        UNION ALL

        SELECT
            __tree.tree_depth + 1 AS tree_depth,
            __tree.tree_path || T.id,
            __tree.tree_ordering || _name::text,
            T."id"
        FROM dcim_location T
        JOIN __tree ON T."parent_id" = __tree.tree_pk
    )
    SELECT (__tree.tree_depth) AS "tree_depth", (__tree.tree_path) AS "tree_path", (__tree.tree_ordering) AS "tree_ordering", "dcim_location"."id", "dcim_location"."created", "dcim_location"."last_updated", "dcim_location"."_custom_field_data", "dcim_location"."status_id", "dcim_location"."parent_id", "dcim_location"."name", "dcim_location"."_name", "dcim_location"."slug", "dcim_location"."location_type_id", "dcim_location"."site_id", "dcim_location"."tenant_id", "dcim_location"."description" FROM "dcim_location" , "__tree" WHERE ((bda87ad3-6e6e-4a28-b32a-77080a470ba9 = ANY(__tree.tree_path)) AND NOT ("dcim_location"."id" = bda87ad3-6e6e-4a28-b32a-77080a470ba9) AND (__tree.tree_pk = dcim_location.id)) ORDER BY ("__tree".tree_ordering) ASC
    ```

Reply:
Thank you!